### PR TITLE
Bug: PS-247 - keyring_vault mtr tests should be skipped if Vault server

### PR DESCRIPTION
**Pull request by robgolebiowski**
is unavailable

Added check file is_vault_server_up.inc. This script checks if
Hashicorp Vault server that is to be used in a test is available. In
case it is not the test will be skipped. This test file was included in
all keyring_vault's MTR tests.

### CODE REVIEW

**Review by percona-ysorokin — COMMENTED** [review]
Just a suggestion: it may be better to make an attempt to establish a connection to `<vault_host>:443` rather than ping the host because our IT guys could decide to filter out ICMP traffic one day and they will do this silently.

File: plugin/keyring_vault/tests/mtr/is_vault_server_up.inc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  $F->close();
+  $p->close();
+EOF
+
+--source $MYSQLTEST_VARDIR/tmp/ping_result.inc
+--remove_file $MYSQLTEST_VARDIR/tmp/ping_result.inc
+
+if(!$PING_SUCCESSFUL)
+{
+  --skip Seems that Hashicorp Vault is down.
```
Comment (laurynas-biveinis) [inline plugin/keyring_vault/tests/mtr/is_vault_server_up.inc]: ... testing server ...

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
+1 to Yura's comment if feasible to implement

**Comment by robgolebiowski** [comment]
It is not really possible to try establishing connection with Vault Server. As it uses REST protocol I can only send https requests that Vault server will understand. I.e. I can list secrets, create or remove secret. Listing secrets is not an option as that would mean retrieving list of all keys from the server - which can be time consuming and just too much for simple check. Also creating and removing a test secret seems like an overkill. I could try using tokens backed to retrieve information about current token that I am using - but this mean upgrading token's policy to the level that is not needed. Token used in test should only be able to create/remove/list secrets from the backend they have access to. I really doubt they will change ICMP routing.

**Comment by robgolebiowski** [comment]
Rebased on the latest 5.7

File: plugin/keyring_vault/tests/mtr/is_vault_server_up.inc (hunk 1 above)
Comment (robgolebiowski) [reply plugin/keyring_vault/tests/mtr/is_vault_server_up.inc]: I am not sure I understand this comment. Should --skip state something like --skip While testing server turned out that Hashicorp Vault is down ?

Comment (laurynas-biveinis) [reply plugin/keyring_vault/tests/mtr/is_vault_server_up.inc]: "Seems that Hashicorp Vault testing server is down"

**Comment by laurynas-biveinis** [comment]
If this has been updated and ready for review, then a new Jenkins run for the affected testcases is needed

**Comment by robgolebiowski** [comment]
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1318/

File: plugin/keyring_vault/tests/mtr/is_vault_server_up.inc (hunk 1 above)
Comment (laurynas-biveinis) [reply plugin/keyring_vault/tests/mtr/is_vault_server_up.inc]: The comment not addressed but too minor to hold the merge on its own

**Review by laurynas-biveinis — APPROVED** [review]

**Review by percona-ysorokin — APPROVED** [review]
LGTM

Good as an initial version. However, still believe that just establishing a TCP connection to `$vault_url` port 443 (even without SSL handshake and sending any HTTP request) could be a good improvement.

**Comment by robgolebiowski** [comment]
I have changed the implementation and addressed (I hope ;) Yura's comment. Please have another look, if necessary I will create separate PR.

**Review by laurynas-biveinis — APPROVED** [review]

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Comment by robgolebiowski** [comment]
JR => https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1395/

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**